src/scala/scripts/TwitterScroogeGenerator.scala

```diff
@@ -0,0 +1,269 @@
+package scripts
+
+import com.twitter.scrooge.Compiler
+
+import scala.collection.mutable.Buffer
+import scala.io.Source
+
+import java.io.{ File, FileOutputStream, IOException }
+import java.nio.file.{ Files, SimpleFileVisitor, FileVisitResult, Path, Paths }
+import java.nio.file.attribute.{ BasicFileAttributes, FileTime }
+import java.util.jar.{ JarEntry, JarFile, JarOutputStream }
+
+object FinalJarCreator {
+  val gm = """(\S+) -> (\S+)""".r
+
+  def apply(dest: String, owned: Set[String], genFileMap: String, scroogeDir: String) {
+    val genmap = Source.fromFile(genFileMap)
+      .getLines
+      .foldLeft(Map.empty[String, Set[String]]) { case (m, gm(thrift, gen)) =>
+        m.+((thrift, m.getOrElse(thrift, Set.empty[String]) + gen))
+      }
+    val shouldMove =
+      owned.foldLeft(Set.empty[String]) { (s, n) =>
+        genmap.get(n).fold(s) { s ++ _ }
+      }.map { Paths.get(_).normalize }
+    val jar = new JarOutputStream(new FileOutputStream(dest))
+    Files.walkFileTree(
+      Paths.get(scroogeDir),
+      FinalJarCreator(scroogeDir, jar, shouldMove)
+    )
+    jar.close()
+  }
+}
+case class FinalJarCreator(_baseDir: String, jar: JarOutputStream, shouldMove: Set[Path]) extends SimpleFileVisitor[Path] {
+  val baseDir = Paths.get(_baseDir).normalize
+
+  // We return the path of the file to add to the jar
+  def shouldVisitFile(file: Path): Option[Path] =
+    if (shouldMove.contains(file)) Some(baseDir.relativize(file))
+    else None
+
+  override def visitFile(file: Path, attr: BasicFileAttributes) = {
+    shouldVisitFile(file).foreach { _file =>
+      val entry = new JarEntry(_file.toString)
+      entry.setTime(198001010000L)
+      jar.putNextEntry(entry)
+      Files.copy(file, jar)
+    }
+    FileVisitResult.CONTINUE
+  }
+}
+
+//TODO deal with errors etc
+object DeleteRecursively extends SimpleFileVisitor[Path] {
+  override def visitFile(file: Path, attr: BasicFileAttributes) = {
+    Files.delete(file)
+    FileVisitResult.CONTINUE
+  }
+
+  override def postVisitDirectory(dir: Path, e: IOException) = {
+    if (e != null) throw e
+    Files.delete(dir)
+    FileVisitResult.CONTINUE
+  }
+}
+
+case class ForeachFile(f: Path => Unit) extends SimpleFileVisitor[Path] {
+  override def visitFile(file: Path, attr: BasicFileAttributes) = {
+    f(file)
+    FileVisitResult.CONTINUE
+  }
+}
+
+//TODO add logging?
+object ScroogeGenerator {
+  def deleteDir(path: Path) {
+    try {
+      Files.walkFileTree(path, DeleteRecursively)
+    } catch {
+      case e: Exception => //TODO LOG
+    }
+  }
+
+  def extractJarTo(_jar: String, _dest: String): List[Path] = {
+    val files = Buffer[Path]()
+    //TODO add to it, error if duplicate!
```

Comment:
looks like we are erroring below, aren't we (it says it throws). What does this mean?